service/history/workflow/context.go

```diff
@@ -698,48 +697,67 @@ func (c *ContextImpl) SubmitClosedWorkflowSnapshot(
 	)
 }
 
-func (c *ContextImpl) mergeContinueAsNewReplicationTasks(
-	updateMode persistence.UpdateWorkflowMode,
+func (c *ContextImpl) mergeUpdateWithNewReplicationTasks(
 	currentWorkflowMutation *persistence.WorkflowMutation,
 	newWorkflowSnapshot *persistence.WorkflowSnapshot,
 ) error {
 
-	if currentWorkflowMutation.ExecutionState.Status != enumspb.WORKFLOW_EXECUTION_STATUS_CONTINUED_AS_NEW {
+	if newWorkflowSnapshot == nil {
 		return nil
-	} else if updateMode == persistence.UpdateWorkflowModeBypassCurrent && newWorkflowSnapshot == nil {
-		// update current workflow as zombie & continue as new without new zombie workflow
-		// this case can be valid if new workflow is already created by resend
+	}
+
+	// TODO: looks like ndc tests will generate this invalid case
+	// if currentWorkflowMutation.ExecutionState.Status == enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING {
```

Comment:
no harm to have this sanity check

Reply:
I can't right now. The random event generator in NDC tests will generate cases violating this sanity check I think. Do you think that's expected? or that actually suggests there's something wrong with my implementation.

Reply:
This sanity check is not valid. The case can happen when standby is turning current running workflow into zombie and creating a new run. The logic here is fine though because I think this only happens in standby cluster so the # of replication tasks for both currentWorkflowMutation and newWorkflowSnapshot are 0 and the function will be a no-op.